tracker/tracker/src/main/modules/input.ts

```diff
@@ -173,18 +157,60 @@ export default function (app: App, opts: Partial<Options>): void {
       // TODO: support multiple select (?): use selectedOptions; Need send target?
       if (hasTag(node, 'select')) {
         sendInputValue(id, node)
-        app.attachEventListener(node, 'change', () => {
+        const handler = () => {
           sendInputValue(id, node)
-        })
+        }
+        node.addEventListener('change', handler)
+        app.attachEventListener(node, 'change', handler, false, true, true)
       }
+
       if (isTextEditable(node)) {
         inputValues.set(id, node.value)
         sendInputValue(id, node)
+        const setFocus = () => {
+          Object.assign(node, { or_focusStart: +new Date() })
+        }
+        const inputEvent = (e: InputEvent) => {
+          const value = (e.target as HTMLInputElement).value
+          if (inputValues.get(id) === '' && value !== '') {
+            const inputTime = +new Date()
+            // @ts-ignore
+            const hesitationTime = inputTime - node.or_focusStart
+            Object.assign(node, { or_inputHesitation: hesitationTime })
+          }
+          inputValues.set(id, value)
+          debouncedUpdate(id, node)
+        }
+        const changeEvent = (e: InputEvent) => {
+          const value = (e.target as HTMLInputElement).value
+          if (inputValues.get(id) !== value) {
+            inputValues.set(id, value)
+            debouncedUpdate(id, node)
+          }
+          Object.assign(node, { or_inputHesitation: undefined, or_focusStart: undefined })
+        }
+        node.addEventListener('focus', setFocus)
+        node.addEventListener('input', inputEvent)
+        node.addEventListener('change', changeEvent)
+        app.attachEventListener(node, 'focus', setFocus, false, true, true)
+        app.attachEventListener(node, 'input', inputEvent, false, true, true)
+        app.attachEventListener(node, 'change', changeEvent, false, true, true)
         return
       }
-      if (isCheckable(node)) {
-        checkableValues.set(id, node.checked)
+
+      if (isCheckbox(node)) {
+        checkboxValues.set(id, node.checked)
         app.send(SetInputChecked(id, node.checked))
+        const checkboxChange = (e: InputEvent) => {
+          const value = (e.target as HTMLInputElement).checked
+          if (checkboxValues.get(id) !== value) {
```

Comment:
This supposed to be always `true`. If you have the `change` event on checkbox that means it has changed. Even if not for some reason (do you have an idea in what case??) - there is no problem to send it twice.

So no need  for `checkboxValue` map here.